openassessment/workflow/models.py

```diff
@@ -376,6 +377,11 @@ def update_from_assessments(
                     if override_submitter_requirements:
                         step.submitter_completed_at = common_now
                     step.save()
+                if self.status == self.STATUS.done:
```

Comment:
Can this logic be moved to line 387 below the already existing  `if self.status == self.STATUS.done:`